src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityAnalysisResultRepository.java

```diff
@@ -95,4 +134,257 @@ public String findStatus(UUID resultUuid) {
+            .isJustBefore(selector.getIsJustBefore())
+            .functionType(selector.getFunctionType())
+            .requestedChunkSize(selector.getChunkSize() == null ? 0 : selector.getChunkSize())
+            .chunkOffset(selector.getOffset() == null ? 0 : selector.getOffset());
+
+        List<ContingencyEmbeddable> cs = sas.getContingencies();
+        List<SensitivityFactorP> fs = sas.getFactors();
+        Collection<String> funcIds = selector.getFunctionIds();
+        Collection<String> varIds = selector.getVariableIds();
+        Collection<String> contingencyIds = selector.getContingencyIds();
+        if (canOnlyReturnEmpty(funcIds, varIds, contingencyIds, selector.getFunctionType(), cs, fs)) {
+            retBuilder.totalSensitivitiesCount(0);
+            retBuilder.filteredSensitivitiesCount(0);
+            retBuilder.sensitivities(List.of());
+            return retBuilder.build();
+        }
+
+        Map<Pair<String, String>, SensitivityOfTo> before = new HashMap<>();
+        Set<String> allFunctionIds = new TreeSet<>();
+        Set<String> allVariableIds = new TreeSet<>();
+        Set<String> allContingencyIds = new TreeSet<>();
+        int count = apply(sas, funcIds, varIds, null, cs, fs, selector.getFunctionType(), selector.getIsJustBefore(),
+            allFunctionIds, allVariableIds, null, (sar, c, f) -> {
+                if (c != null) {
+                    return;
+                }
+                before.put(Pair.of(f.getFunctionId(), f.getVariableId()), SensitivityOfTo.builder()
+                    .funcId(f.getFunctionId())
+                    .varId(f.getVariableId())
+                    .varIsAFilter(f.isVariableSet())
+                    .value(sar.getValue())
+                    .functionReference(sar.getFunctionReference())
+                    .build());
+            });
+
+        if (selector.getIsJustBefore()) {
+            Comparator<SensitivityOfTo> cmp = makeComparatorOfTo(selector.getSortKeysWithWeightAndDirection());
+            List<SensitivityOfTo> befores = new ArrayList<>(before.values());
+            if (cmp != null) {
+                befores.sort(cmp);
+            }
+            complete(retBuilder, allFunctionIds, allVariableIds, null, count, selector, befores);
+            return retBuilder.build();
+        } else {
+            List<SensitivityWithContingency> after = new ArrayList<>();
+            count = apply(sas, funcIds, varIds, contingencyIds, cs, fs, selector.getFunctionType(), false,
+                allFunctionIds, allVariableIds, allContingencyIds, (sar, c, f) -> {
+                    if (c == null) {
+                        return;
+                    }
+                    SensitivityOfTo b = before.get(Pair.of(f.getFunctionId(), f.getVariableId()));
+                    SensitivityWithContingency r = SensitivityWithContingency.toBuilder(b)
```

Comment:
toBuilder(b) here will first set funcId, varId ... and then you owerwrite these fields with f object fields
so why using toBuilder ?

Reply:
You are right !